File: python_programmers_school/level 1/음양 더하기.py
def solution(absolutes, signs):
    z = list(zip(absolutes, signs))
    result = 0
    for i in z:
        if i[1] == True:
            result += i[0]
        else:
            result += i[0] * -1
    return result

# 조금 더 간단히 적기
# def solution(absolutes, signs):
#     answer=0
#     for absolute,sign in zip(absolutes,signs):     --> absolute와 sign에 (absolutes, sign) 안의 요소를 하나씩 차례로 할당
#         if sign:
#             answer+=absolute
#         else:
#             answer-=absolute     --> * -1 할 필요 없이 -=로 해주면 간단
#     return answer

File: javascript_programmers_school/level 1/음양 더하기.js
function solution(absolutes, signs) {
  let result = 0;
  for (let i = 0; i < signs.length; i++) {
    signs[i] ? (result += absolutes[i]) : (result += -absolutes[i]);
  }
  return result;
}

// 다른 사람의 풀이
// reduce를 이용해 깔끔하게 작성하기
function solution(absolutes, signs) {
  return absolutes.reduce((acc, val, i) => acc + val * (signs[i] ? 1 : -1), 0);
}
